src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -415,7 +415,11 @@ final void doExecutePreparedStatement(PrepStmtExecCmd command) throws SQLServerE
 
         boolean hasNewTypeDefinitions = buildPreparedStrings(inOutParam, false);
         if ((Util.shouldHonorAEForParameters(stmtColumnEncriptionSetting, connection)) && (0 < inOutParam.length) && !isInternalEncryptionQuery) {
-            getParameterEncryptionMetadata(inOutParam);
+            
+            // retrieve paramater encryption metadata if they are not retrieved yet
+            if (null == inOutParam[0].getCryptoMetadata()) {
+                getParameterEncryptionMetadata(inOutParam);
+            }
```

Comment:
this fix will work only if the first parameter is encrypted. Consider the case where first parameter is not encrypted, then it will fall back to the old behavior as `inOutParam[0].getCryptoMetadata()` is null.